mlir/include/mlir/Dialect/XeGPU/Transforms/Transforms.h

```diff
@@ -12,11 +12,18 @@
 namespace mlir {
 class RewritePatternSet;
 
+namespace vector {
+struct UnrollVectorOptions;
+} // namespace vector
+
 namespace xegpu {
 
 /// Appends patterns for folding aliasing ops into XeGPU ops into `patterns`.
 void populateXeGPUFoldAliasOpsPatterns(RewritePatternSet &patterns);
 
+void populateXeGPUUnrollPatterns(RewritePatternSet &patterns,
+                                 const vector::UnrollVectorOptions &options);
```

Comment:
It's better to decouple from the generic vector infrastructure and create XeGPU specific options (even if they end up being the same).

Reply:
> It's better to decouple from the generic vector infrastructure and create XeGPU specific options (even if they end up being the same).

Even these patterns are to be used together with vector unroll patterns in the same pass?

Reply:
@adam-smnk decoupled it as you suggested.